### PR TITLE
Double check after regex search for cudnn_path in configure.py

### PR DESCRIPTION
**Pull request by ScorpioCPH**
If chose wrong `cuDNN` version on `linux`, `configure.py` will throw an exception as shown below:

```shell
Please specify the location where cuDNN 6 library is installed. Refer to README.md for more details. [Default is /usr/local/cuda]:

Traceback (most recent call last):
  File "configure.py", line 1021, in <module>
    main()
  File "configure.py", line 997, in main
    set_tf_cunn_version(environ_cp)
  File "configure.py", line 688, in set_tf_cunn_version
    cudnn_path_from_ldconfig).group(1)
AttributeError: 'NoneType' object has no attribute 'group'
```

This is because these code return `None`:

```python
cudnn_path_from_ldconfig = re.search('.*libcudnn.so .* => (.*)',
                                     cudnn_path_from_ldconfig)
```

This PR add double check after regex search.

Related PR: https://github.com/tensorflow/tensorflow/pull/12347 

@wangqr @yifeif PTAL.

### CODE REVIEW

File: configure.py [hunk 1]
```diff
@@ -685,7 +685,9 @@ def set_tf_cunn_version(environ_cp):
       ldconfig_bin = which('ldconfig') or '/sbin/ldconfig'
       cudnn_path_from_ldconfig = run_shell([ldconfig_bin, '-p'])
       cudnn_path_from_ldconfig = re.search('.*libcudnn.so .* => (.*)',
-                                           cudnn_path_from_ldconfig).group(1)
+                                           cudnn_path_from_ldconfig)
+      if cudnn_path_from_ldconfig != None:
```
Comment (yifeif) [inline configure.py]: [nit]: remove " != None"

Comment (ScorpioCPH) [reply configure.py]: @yifeif Sorry, I can't see the comments, could you have a check please?

**Comment by wangqr** [comment]
Maybe the next [if block](https://github.com/ScorpioCPH/tensorflow/blob/ab635bcafaa42f599e49ceb6ed324e32996d8b45/configure.py#L691-L693) should only run when `cudnn_path_from_ldconfig` is not None, since some file named `None.6.0` is not what we want

**Comment by ScorpioCPH** [comment]
@wangqr Yes, but before this block, exception has been thrown out on calling `.group(1)`

**Comment by wangqr** [comment]
Yes I cam confirm that we should check for the None value returned by `search`, and your commit does the check. I just suggest that maybe moving the following if block into the if block you've created is more proper.

**Comment by wangqr** [comment]
From
```python
      if cudnn_path_from_ldconfig != None:
        cudnn_path_from_ldconfig = cudnn_path_from_ldconfig.group(1)
      if os.path.exists('%s.%s' % (cudnn_path_from_ldconfig, tf_cudnn_version)):
        cudnn_install_path = os.path.dirname(cudnn_path_from_ldconfig)
        break
```
to
```python
      if cudnn_path_from_ldconfig != None:
        cudnn_path_from_ldconfig = cudnn_path_from_ldconfig.group(1)
        if os.path.exists('%s.%s' % (cudnn_path_from_ldconfig, tf_cudnn_version)):
          cudnn_install_path = os.path.dirname(cudnn_path_from_ldconfig)
          break
```

**Comment by ScorpioCPH** [comment]
@wangqr PR updated, could you take another look, please?

**Review by wangqr — APPROVED** [review]

**Comment by wangqr** [comment]
@ScorpioCPH I don't have write access to this repo...

File: configure.py (hunk 1 above)
Comment (yifeif) [reply configure.py]: Could you remove the "!= None"?

**Comment by yifeif** [comment]
@ScorpioCPH let me know if you still cannot see the review comment.

**Comment by ScorpioCPH** [comment]
@yifeif I see your comment now, thanks! And fix it by amending last commit. PTAL.

**Review by yifeif — APPROVED** [review]

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please

**Comment by yifeif** [comment]
@tensorflow-jenkins test this please

**Pull request merged**